packages/pieces/community/google-drive/src/lib/triggers/new-file.ts

```diff
@@ -57,24 +65,73 @@ export const newFile = createTrigger({
     });
   },
   run: async (context) => {
-    return await pollingHelper.poll(polling, {
+    const newFiles = await pollingHelper.poll(polling, {
       auth: context.auth,
       store: context.store,
       propsValue: context.propsValue,
     });
+
+    const newFilesObj = JSON.parse(JSON.stringify(newFiles))
+
+    for (let i = 0; i < newFilesObj.length; i++) {
+      newFilesObj[i].link = ''
+    }
+
+    if (context.propsValue.getFileContent) {
+      const fileContentPromises: Promise<string>[] = []
+      for (const file of newFilesObj) {
+        fileContentPromises.push(downloadFileFromDrive(context.auth, context.files, file["id"], file["name"]));
+      }
+
+      const filesContent = await Promise.all(fileContentPromises)
+
+      for (let i = 0; i < newFilesObj.length; i++) {
+        newFilesObj[i].link = filesContent[i]
+      }
+
+      return newFilesObj
+    }
+    else {
+      return newFilesObj
+    }
   },
   test: async (context) => {
-    return await pollingHelper.test(polling, {
+    const newFiles = await pollingHelper.test(polling, {
       auth: context.auth,
       store: context.store,
       propsValue: context.propsValue,
     });
+
+    const newFilesObj = JSON.parse(JSON.stringify(newFiles))
+
+    for (let i = 0; i < newFilesObj.length; i++) {
+      newFilesObj[i].link = ''
```

Comment:
Let's call it content, I think better.

for link people might confuse it with shareable link from google.